src/client/Client.cc

```diff
@@ -7956,6 +7956,12 @@ int Client::readlinkat(int dirfd, const char *relpath, char *buf, loff_t size, c
     return r;
   }
 
+  if (!strcmp(relpath, "")) {
+    if (!dirinode.get()->is_symlink())
+      return -CEPHFS_ENOENT;
```

Comment:
This is from readlinkat(2):

```
       EINVAL The named file (i.e., the final filename component of pathname) is not a symbolic link.
```

Reply:
This is the special case of empty pathname. `EINVAL` is returned when the final filename component of pathname is not a symbolic link. But here we don't even have a pathname. Accordingly _errno_ changes to `ENOENT` as we see with standard local file systems based on the following line from https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/fs/stat.c#n554:

```
error = (name->name[0] == '\0') ? -ENOENT : -EINVAL;
```